purchase_cancel_qty/models/purchase_order.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = 'purchase.order'
+
+    @api.multi
+    def button_approve(self, force=False):
+        res = super(PurchaseOrder, self).button_approve(force=force)
+        for rec in self:
+            rec._update_lines_ordered_qty()
+        return res
+
+    @api.multi
+    def _update_lines_ordered_qty(self):
+        self.ensure_one()
```

Comment:
@benwillig Why `self.ensure_one()` ? You could iterate here instead of calling the method once for each record.